adapters/adkernel/adkernel.go

```diff
@@ -232,19 +259,45 @@ func (adapter *adkernelAdapter) MakeBids(internalRequest *openrtb2.BidRequest, e
 	bidResponse.Currency = bidResp.Cur
 	for i := 0; i < len(seatBid.Bid); i++ {
 		bid := seatBid.Bid[i]
+		origImpId := bid.ImpID
+		typeSuffix := ""
+		if strings.HasSuffix(origImpId, mf_suffix) {
+			sfxStart := len(origImpId) - len(mf_suffix) - 1
+			typeSuffix = origImpId[sfxStart:]
+			bid.ImpID = origImpId[:sfxStart]
+		}
 		bidResponse.Bids = append(bidResponse.Bids, &adapters.TypedBid{
 			Bid:     &bid,
-			BidType: getMediaTypeForImpID(bid.ImpID, internalRequest.Imp),
+			BidType: getMediaTypeForImpID(bid.ImpID, typeSuffix, internalRequest.Imp),
 		})
 	}
 	return bidResponse, nil
 }
 
 // getMediaTypeForImp figures out which media type this bid is for
-func getMediaTypeForImpID(impID string, imps []openrtb2.Imp) openrtb_ext.BidType {
-	for _, imp := range imps {
-		if imp.ID == impID && imp.Banner != nil {
+func getMediaTypeForImpID(impID string, typeSuffix string, imps []openrtb2.Imp) openrtb_ext.BidType {
+	if len(typeSuffix) > 0 {
+		if typeSuffix == mf_suffix_banner {
 			return openrtb_ext.BidTypeBanner
+		} else if typeSuffix == mf_suffix_video {
+			return openrtb_ext.BidTypeVideo
+		} else if typeSuffix == mf_suffix_audio {
+			return openrtb_ext.BidTypeAudio
```

Comment:
as per `static/bidder-info/adkernel.yaml` audio format is not supported. Should remove `mf_suffix_audio` check or update `static/bidder-info/adkernel.yaml` to support audio format